File: Prácticas/P3/Ruby/civitas/lib/test_p1.rb
# encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "dado.rb"
require_relative "mazo_sorpresas.rb"
require_relative "civitas_juego.rb"
require_relative "casilla.rb"
require_relative "jugador.rb"
require_relative "titulo_propiedad.rb"

module Civitas
  class TestP1
    NUM_JUGADORES = 4
    @@dado = Dado.instance
    @@mazo = MazoSorpresas.new
    
    def self.get_nombre_jugadores()
      nombres = Array.new
      nombre = ""
      
      for i in 1..NUM_JUGADORES
        print "Nombre del jugador #{i}: "
        nombre = gets.chomp
        nombres << nombre
      end
      nombres
    end
    
    def self.main
      @nombres = get_nombre_jugadores()
      @juego = CivitasJuego.new(@nombres)
      
      @juego.get_jugador_actual.mover_a_casilla(@@dado.tirar + @juego.get_jugador_actual.num_casilla_actual)
      print "Valor del dado #{@@dado.ultimo_resultado}"
      
      #@juego.get_jugador_actual.comprar(@juego.get_casilla_actual.titulo_propiedad)
      #@juego.construir_casa(8)
      
      print "#{@juego.info_jugador_texto}"
      print "#{@juego.get_casilla_actual.to_s}"
      
      @juego.get_jugador_actual.mover_a_casilla(@@dado.tirar+@juego.get_jugador_actual.num_casilla_actual)
      print "Valor del dado #{@@dado.ultimo_resultado}"
      
      print "#{@juego.info_jugador_texto}"
      print "#{@juego.get_casilla_actual.to_s}"
    end
  end
  TestP1.main
end

File: Prácticas/P3/Java/civitas/src/civitas/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Scanner;
import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class TestP1 {

    /**
     * @param args the command line arguments
     */
    
    private static final int numJugadores = 4;
    private static Dado dado = Dado.getInstance();
    private static final Scanner in = new Scanner(System.in);
    
    private static ArrayList<String> getNombreJugadores(){
        ArrayList<String>nombres = new ArrayList();
        String nombre;
        
         for(int i = 0; i < numJugadores; ++i){
            System.out.print("Introduce el nombre del jugador " + (i+1) + ": ");
            nombre = in.nextLine();
            nombres.add(nombre);
        }
        System.out.println();
        
        return nombres;
    }
    
    public static void main(String[] args) {
        ArrayList<String>nombres = getNombreJugadores();
        CivitasJuego juego = new CivitasJuego(nombres);
        
        juego.getJugadorActual().moverACasilla(dado.tirar() + juego.getJugadorActual().getNumCasillaActual());
        System.out.print("\nValor del dado: "+dado.getUltimoResultado());
        
        juego.getJugadorActual().comprar(juego.getCasillaActual().getTituloPropiedad());
        juego.construirCasa(8);
        
        System.out.print("\n"+juego.infoJugadorTexto());
        System.out.print("\n"+juego.getCasillaActual().toString());
        
        juego.getJugadorActual().moverACasilla(dado.tirar() + juego.getJugadorActual().getNumCasillaActual());
        System.out.print("\nValor del dado: "+dado.getUltimoResultado());
        
        System.out.print("\n"+juego.infoJugadorTexto());
        System.out.print("\n"+juego.getCasillaActual().toString());
        
        juego.getJugadorActual().encarcelar(10);
        System.out.print("\nEncarcelado");
        
        System.out.print("\n"+juego.infoJugadorTexto());
        System.out.print("\n"+juego.getCasillaActual().toString());
    }
    
}
